chain/chain/src/chain.rs

```diff
@@ -3781,9 +3785,12 @@ impl Chain {
                 // we can't use hash from the current block here yet because the incoming receipts
                 // for this block is not stored yet
                 let new_receipts = collect_receipts(incoming_receipts.get(&shard_id).unwrap());
+                let shard_layout =
+                    self.epoch_manager.get_shard_layout_from_prev_block(prev_hash)?;
```

Comment:
`get_shard_layout(epoch_id)` is an alternative (if you happen to have the epoch id handy e.g. in the block header)